### PR TITLE
Merge ForecastFactory into CercForecastService

### PR DESCRIPTION
**Pull request by jdudley1123**
## Context
Currently we have two classes, the CercForecastService and the ForecastFactory. CercForecastService is responsible for returning the forecasts from the cache, and refreshing the cache when it is stale. It calls the CERC API and then passes the response to the ForecastFactory.

CercForecastService is the only user of ForecastFactory, and is a very small class, so there is no good reason to have separate class for what is essentially core functionality of the service – building a forecast.

## Changes in this PR
This PR merges ForecastFactory into CercForecastService. It also improves the tests for CercForecastService so that they test behaviour and not implementation.

### CODE REVIEW

File: spec/services/cerc_forecast_service_spec.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 RSpec.describe CercForecastService do
   describe "::latest_forecasts_for(zone)" do
-    let(:zone) { FactoryBot.build(:zone, name: "Barnet") }
+    let(:zone) { create(:zone, cerc_id: 29) }
```
Comment (patrickjfl) [inline spec/services/cerc_forecast_service_spec.rb]: if we're being consistent here, should we leave it as `FactoryBot.build`?

**Review by patrickjfl — APPROVED** [review]
Nice one bringing ForecastFactory into the service 👍 

I can see the benefits of testing behaviour over implementation, but also the drawbacks. Does it not make our tests harder to debug when something changes, causing them to fail?

File: spec/services/cerc_forecast_service_spec.rb (hunk 1 above)
Comment (jdudley1123) [reply spec/services/cerc_forecast_service_spec.rb]: Good spot!

**Comment by jdudley1123** [comment]
> Nice one bringing ForecastFactory into the service 👍
> 
> I can see the benefits of testing behaviour over implementation, but also the drawbacks. Does it not make our tests harder to debug when something changes, causing them to fail?

If your tests are tightly coupled to your implementation, then you have to change them in lockstep and it becomes difficult to tell whether the change you made has changed the app's behaviour because you've also changed the tests that check that. If the tests are focused on behaviour, then changing the implementation (for example to use a new algorithm) should not mean there are any significant changes to the tests.

Here's a good article about it that actually mentions caches as an exception: https://testing.googleblog.com/2013/08/testing-on-toilet-test-behavior-not.html

**Pull request merged**